corehq/apps/cleanup/tasks.py

```diff
@@ -253,7 +254,7 @@ def delete_web_user():
         ]:
             web_user = WebUser.get_by_username(username)
             if web_user:
-                web_user.delete(deleted_by=None, deleted_via=__name__)
+                web_user.delete(None, deleted_by=SYSTEM_USER_ID, deleted_via=__name__ + '.delete_web_user')
```

Comment:
won't this fail the `if not domain` validation?

Reply:
I see you changed the method to allow `None` for domain if the user is `SYSTEM_USER_ID`

Reply:
Why not support `deleted_by` being None for system changes instead of checking for `SYSTEM_USER_ID`? It's weird that `deleted_by` can be either a user or a string, and it's unfortunate to add that bit of complexity that's only used by this one caller.

Tangentially, this is a weird task. The original PR (https://github.com/dimagi/commcare-hq/pull/23551) says this task was requested by QA. @Ben-Talbot do you know what this task is and if it's still needed? It checks every night if there's a web user named either `create_growth<at>outlook.com` or `growth_analytics<at>outlook.com`, and if it finds such a user, deletes it.

Reply:
> Why not support deleted_by being None for system changes instead of checking for SYSTEM_USER_ID?

You mean assume that if there is no deleted by, it was system. I am not sure I like that, being explicit is good.
But also, currently we recognize it's a system change eventually from the `deleted_by` passed, so if that is not passed, we wont be able to identify such changes.

>  can be either a user or a string

I think HQ treats `SYSTEM_USER_ID` as an ID [along with some other ones](https://github.com/dimagi/commcare-hq/blob/f8f382e06a90820dc44d21998da9a95927f7af9d/corehq/pillows/utils.py#L59). Its the same thing we have with forms as well.

>  bit of complexity that's only used by this one caller.

Its true that this is one instance and would be great if we can stop doing it. Though I think we would have more such system updates to user in future PR so would be good to agree on a way to log updates to users by SYSTEM_USER_ID.

Reply:
I do agree with you about it being good to be explicit. I'm surprised there aren't other places where we have the system changing users. You've gone through all of the code, right? Maybe we just don't often have the system *delete* users.

> I think HQ treats SYSTEM_USER_ID as an ID

Even though it's a known ID, it's still a string, so `changed_by_user` is either a string or a couch document, which are two very different types to be represented by the same variable.